engine/table/src/main/java/io/deephaven/engine/util/PythonScopeJpyImpl.java

```diff
@@ -89,30 +93,106 @@ public Object convertValue(PyObject value) {
      * varargs method, so that we can call it using __call__ without all of the JPy nastiness.
      */
     public static class CallableWrapper {
+        private static final Map<Character, Class<?>> numpyType2JavaClass = new HashMap<>();
+
+        static {
+            numpyType2JavaClass.put('i', int.class);
+            numpyType2JavaClass.put('l', long.class);
+            numpyType2JavaClass.put('h', short.class);
+            numpyType2JavaClass.put('f', float.class);
+            numpyType2JavaClass.put('d', double.class);
+            numpyType2JavaClass.put('b', byte.class);
+            numpyType2JavaClass.put('?', boolean.class);
+            numpyType2JavaClass.put('O', Object.class);
+        }
+
         private final PyObject pyObject;
 
+        private String signature = null;
+        private List<Class<?>> paramTypes;
+        private Class<?> returnType;
+        private boolean vectorizable = false;
+        private boolean vectorized = false;
+
+        public void setVectorizable(boolean vectorizable) {
+            this.vectorizable = vectorizable;
+        }
+
+        public boolean isVectorizable() {
+            return vectorizable;
+        }
+
+
         public CallableWrapper(PyObject pyObject) {
             this.pyObject = pyObject;
         }
 
+        private void prepareSignature() {
+            if (pyObject.getType().equals(NUMBA_VECTORIZED_FUNC_TYPE)) {
+                List<PyObject> params = pyObject.getAttribute("types").asList();
+                if (params.isEmpty()) {
+                    throw new IllegalArgumentException("numba vectorized function must have an explicit signature.");
```

Comment:
Do we provide enough info for a user to figure out what function is causing the problem?  e.g. does the stack have the function name or anything else useful in locating the problem?

Reply:
added string representation of pyObject to the exception message